crates/miden-lib/asm/miden/input_note.masm

```diff
@@ -38,3 +42,93 @@ export.get_assets_info
     end
     # => [ASSETS_COMMITMENT, num_assets]
 end
+
+#! Writes the assets of the input note with the specified index into memory starting at the 
+#! specified address.
+#!
+#! Attention: memory starting from the `dest_ptr` should have enough space to store all the assets 
+#! in specified note. Make sure that at least `4 * assets_num` memory elements are available.
+#!
+#! The memory layout after the execution of this procedure will look like so:
+#! [ASSET_0, ASSET_1, ..., ASSET_N], where each asset occupies one word. For more detailed
+#! information about the layout of each asset see the description of the `Asset` enum in 
+#! `miden-objects/src/asset/mod.rs` file.
+#!
+#! Inputs:  [note_index, dest_ptr]
+#! Outputs: [num_assets, note_index, dest_ptr]
```

Comment:
Thanks for expanding the docs!

Nit: Since callers don't know `num_assets` yet before calling this function they also can't reserve enough space in `dest_ptr`. So I think we should default to saying:

> Attention: memory starting from the `dest_ptr` should have enough space to store all the assets 
> in the specified note. Make sure that at least `4 * MAX_ASSETS_PER_NOTE` memory elements are available.